CDP4Reporting/DataSource/ReportingDataSourceNode.cs

```diff
@@ -203,6 +208,22 @@ internal static DataTable GetTable(CategoryHierarchy categoryHierarchy)
             }
         }
 
+        /// <summary>
+        /// Checks if the associated <see cref="ElementBase"/> matches the given <see cref="Category"/>.
+        /// TODO check in supercategories also
```

Comment:
please make sure when you add a TODO in code there is a corresponding GH ticket so we do not forget to implement it.

Furthermore, what is the intention of matching? I provide some context which may let you decide to rename the method.

an ICategorizable thing can be made a member of a Category, Categories have subcategories and supercategories. When asserting whether an ICategorizable  is a member of a category the super/sub classing in the category model needs to be taken into account. There are methods in the SDK that do this. My impression is that in the code below we are only doing the assert for a specific category, not taking the complete category model into account.

Reply:
@samatrhea This branche is not going to be merged, because of the architectural difference compared to another branch.
In that branch the SDK is indeed used for category membership checking.